### PR TITLE
Fixed #30597 -- Clarified how to unapply migrations.

### PR DESCRIPTION
**Pull request by davidmgvaz**
I think the rewrite help clarify migrate [app_label] zero. As well as it alerts for possible reversions on other dependent migrations from other apps.

### CODE REVIEW

File: docs/ref/django-admin.txt [hunk 1]
```diff
@@ -805,6 +805,20 @@ The behavior of this command changes depending on the arguments provided:
   migrated past the named migration. Use the name ``zero`` to unapply all
   migrations for an app.
```
Comment (felixxm) [inline docs/ref/django-admin.txt]: I would change a wording here to proposed in the ticket, e.g.
```
    Use the name ``zero`` to migrate all the way back i.e. to revert all
    applied migrations for an app.
```

File: docs/ref/django-admin.txt [hunk 2]
```diff
@@ -805,6 +805,20 @@ The behavior of this command changes depending on the arguments provided:
...
 
+.. admonition:: ``django-admin migrate <app_label> zero``
+
+    The special zero ``<migration_name>`` to unapply(reverse) all the applied
+    migrations for the given ``<app_label>``.
```
Comment (felixxm) [inline docs/ref/django-admin.txt]: I think we should add a short section in `docs/topics/migrations/` instead of this admonition, e.g.
```
.. _migration-zero:

Reverting migrations
================

If you want to revert all migrations applied for an app ...
```

File: docs/ref/django-admin.txt [hunk 3]
```diff
@@ -805,6 +805,20 @@ The behavior of this command changes depending on the arguments provided:
...
+
+.. warning::
+
+    When unapplying any migration, all dependent migrations will also be
+    unapplied (reverted), regardless of ``<app_label>``. This is also truth
+    for ``migrate <app_label> zero``, so use it with caution.
+
+    You can always use ``--plan`` to check the migration plan and and prevent
+    any unwanted propagation to other apps.
```
Comment (felixxm) [inline docs/ref/django-admin.txt]: This warning is not a part of the ticket solution (but I agree that it's a good improvement), please move it to a separate commit. I also think that we don't need a special warning for a `zero` migration, so maybe:
```
.. warning:

    Be aware that when unapplying migrations all dependent migrations will also
    be reverted, regardless of ``<app_label>``. You can use ``--plan`` to check
    which migrations will be reverted.
```

Comment (davidmgvaz) [reply docs/ref/django-admin.txt]: Hi just created a new PR just for this warning pull/11842

Comment (felixxm) [reply docs/ref/django-admin.txt]: Thanks!

File: docs/ref/django-admin.txt (hunk 2 above)
Comment (davidmgvaz) [reply docs/ref/django-admin.txt]: Hi, just updated de PR. Hope it is clear.

**Comment by felixxm** [comment]
@davidmgvaz Thanks :+1: Looks good, I just pushed minor edits and moved _"Reverting migrations"_ to main sections.

**Comment by davidmgvaz** [comment]
And is it good to go right?

**Comment by felixxm** [comment]
Yes, I'm just waiting for tests.

**Comment by davidmgvaz** [comment]
Another thing. There is no piece of documentation or help stating migration_name can be a prefix, this is actually useful but hard to get without docs. Should I open a ticket or just create a PR?

**Comment by felixxm** [comment]
@davidmgvaz It looks like sth that can be added to [django-admin/#migrate](https://docs.djangoproject.com/en/dev/ref/django-admin/#migrate). For such a small cleanup ticket is not required.

**Comment by davidmgvaz** [comment]
Just submitted the PR pull/11846 about migration_name prefix funcionality
